### ISSUE TITLE
DeviceInfo.getMostRecentLocation crash on Meizu

### ISSUE THREAD

**Issue by acbelter** [amplitude/Amplitude-Android#166]
```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke interface method 'java.util.Iterator java.util.List.iterator()' on a null object reference
       at android.os.Parcel.readException(Parcel.java:1690)
       at android.os.Parcel.readException(Parcel.java:1637)
       at android.location.ILocationManager$Stub$Proxy.getLastLocation(ILocationManager.java:754)
       at android.location.LocationManager.getLastKnownLocation(LocationManager.java:1225)
       at com.amplitude.api.DeviceInfo.getMostRecentLocation(DeviceInfo.java:354)
       at com.amplitude.api.AmplitudeClient.logEvent(AmplitudeClient.java:910)
       at com.amplitude.api.AmplitudeClient$5.run(AmplitudeClient.java:841)
       at android.os.Handler.handleCallback(Handler.java:751)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:154)
       at android.os.HandlerThread.run(HandlerThread.java:61)

```
Meizu M6 Note, Meizu PRO 7
Android 7.0-7.1.2

**Comment by ipereziriarte** [comment]
This is happening in General Mobile Discovery II mini and Samsung galaxy s6 edge.
Both in Android 4 & 7

```
Fatal Exception: java.lang.NullPointerException
       at android.os.Parcel.readException(Parcel.java:1486)
       at android.os.Parcel.readException(Parcel.java:1429)
       at android.location.ILocationManager$Stub$Proxy.getLastLocation(ILocationManager.java:651)
       at android.location.LocationManager.getLastKnownLocation(LocationManager.java:1173)
       at com.amplitude.api.DeviceInfo.getMostRecentLocation(SourceFile:354)
       at com.amplitude.api.AmplitudeClient.logEvent(SourceFile:910)
       at com.amplitude.api.AmplitudeClient$5.run(SourceFile:841)
       at android.os.Handler.handleCallback(Handler.java:808)
       at android.os.Handler.dispatchMessage(Handler.java:103)
       at android.os.Looper.loop(Looper.java:193)
       at android.os.HandlerThread.run(HandlerThread.java:61)
```

**Comment by kollesnica-power** [comment]
Same crash on Meizu M6 Note(Android 7.1.2). 

Any ideas?

**Comment by enginegl** [comment]
Same crash on Meizu.
Models: 16th, M1852, M6.
Android versions: 5, 7, 8.

This issue was created almost 3 month ago. 
Can you please add simple try-catch? It's the right way to solve this case.

**Comment by djih** [comment]
Thanks for reporting, this is fixed in v2.23.2: https://github.com/amplitude/Amplitude-Android/releases/tag/v2.23.2
